src/constants.ts

```diff
@@ -36,12 +73,36 @@ export const ROID_SPAWN_TIME = 1; // One asteroid every three seconds
 export const STARTING_SCORE = 0;
 export const DEBUG = false; // Show ship collision boundary and ship center dot
 export const SAVE_KEY_HIGH_SCORE = 'highscore'; // localstorage of high score.
-export const SAVE_KEY_SOUND_ON = 'musicOn'; // localstorage of high score.
-export const SAVE_KEY_MUSIC_ON = 'soundOn'; // localstorage of high score.
 export const NEXT_LEVEL_POINTS = 1000;
 
 /* Drawing Constants*/
 export const TEXT_SIZE = 40; // Text font height in pixels
 export const TEXT_FADE_TIME = 2.5; // text fade in seconds.
-export const CVS = document.querySelector('canvas');
-export const CTX = CVS.getContext('2d');
+const _CVS = document.querySelector('canvas');
+if (!_CVS) throw new Error("Couldn't find canvas element");
+const _CTX = _CVS.getContext('2d');
+if (!_CTX) throw new Error("Couldn't obtain canvas context");
+
+export const CVS = _CVS;
+export const CTX = _CTX;
+
+let ROID_NUM: number;
+function setDifficulty(difficulty: string): void {
+  switch (difficulty) {
+    case 'easy':
+      ROID_NUM = 5;
+      break;
+    case 'medium':
+      ROID_NUM = 10;
+      break;
+    case 'hard':
+      ROID_NUM = 50;
+      break;
+  }
+}
+
+function getRoidNum(): number {
+  return ROID_NUM;
+}
```

Comment:
I like this way of using ENUMs to lock down the string that's able to be passed in. I wonder if there is a way for me to avoid mutating `difficulty` at all.

Reply:
Yep, it's possible with smth like Redux, but this would be an overkill 😄

I think it's ok to have such a mutation because the variable is sealed in a module with getters and setters. The approach is similar to having a private class property without direct manipulation.